src/graph/serialize/heterograph_serialize.cc

```diff
@@ -0,0 +1,297 @@
+/*!
+ *  Copyright (c) 2019 by Contributors
+ * \file graph/serialize/heterograph_serialize.cc
+ * \brief DGLHeteroGraph serialization implementation
+ *
+ * The storage structure is
+ * {
+ *   // MetaData Section
+ *   uint64_t kDGLSerializeMagic
+ *   uint64_t kVersion = 2
+ *   uint64_t GraphType = kDGLHeteroGraph
+ *   ** Reserved Area till 4kB **
+ *
+ *   dgl_id_t num_graphs
+ *   vector<dgl_id_t> graph_indices (start address of each graph)
```

Comment:
Is this still accurate? You probably need to add some descriptions about how to read index data stored at the end of the file.